packages/xrpl/src/sugar/autofill.ts

```diff
@@ -88,6 +97,42 @@ async function autofill<T extends Transaction>(
   return Promise.all(promises).then(() => tx)
 }
 
+async function setNetworkID(client: Client, tx: Transaction): Promise<void> {
+  if (client.networkID === undefined) {
+    await setNetworkIDAndBuildVersion(client)
+  }
+  if (
+    client.networkID !== undefined &&
+    client.networkID > RESTRICTED_NETWORKS
+  ) {
+    // autofill transaction's networkID if either the network is hooks testnet or build version is >= 1.11.0
+    if (
+      (client.buildVersion &&
+        isEarlierRippledVersion(
+          REQUIRED_NETWORKID_VERSION,
+          client.buildVersion,
+        )) ||
+      client.networkID === HOOKS_TESTNET_ID
+    ) {
+      // eslint-disable-next-line no-param-reassign -- param reassign is safe
+      tx.NetworkID = client.networkID
+    } else {
+      // eslint-disable-next-line no-param-reassign -- param reassign is safe
+      tx.NetworkID = undefined
+    }
+  }
+}
+
+async function setNetworkIDAndBuildVersion(client: Client): Promise<void> {
+  const response: ServerInfoResponse = await client.request({
+    command: 'server_info',
+  })
+  // eslint-disable-next-line no-param-reassign, require-atomic-updates -- param reassign is safe, no race condition
+  client.networkID = response.result.info.network_id ?? undefined
+  // eslint-disable-next-line no-param-reassign, require-atomic-updates -- param reassign is safe, no race condition
+  client.buildVersion = response.result.info.build_version
```

Comment:
This would have a race condition if the network could actually change on a transaction by transaction basis. 

If you were sending a transaction to a sidechain & then sending another transaction to mainnet, you could end up with one of those updating the networkID _while_ the other was also being autofilled, and thus have the wrong networkID since it's shared state. (That's not a real scenario though because as it stands a Client only connects to one network)

(Setting the value once on `connect` fixes this - just noting for awareness :) )

Reply:
I see what you meant but if a `client` connect to sidechain then send a transaction to mainnet then the networkId will be incorrect anyway (which is not a real scenario with the current setup anyway)